control-plane/subcommand/server-acl-init/command.go

```diff
@@ -527,14 +526,22 @@ func (c *Command) Run(args []string) int {
 			return 1
 		}
 
+		serviceAccountName := c.withPrefix("connect-injector")
+		componentAuthMethodName := localComponentAuthMethodName
+
 		// If namespaces are enabled, the policy and token need to be global
 		// to be allowed to create namespaces.
 		if c.flagEnableNamespaces {
-			err = c.createGlobalACL("connect-inject", injectRules, consulDC, primary, consulClient)
+			// Create the controller ACL Policy, Role and BindingRule but do not issue any ACLTokens or create Kube Secrets.
+			// ConnectInjector token must be global when namespaces are enabled. This means secondary datacenters need
+			// a token that is known by the primary datacenters.
+			if !primary {
+				componentAuthMethodName = globalComponentAuthMethodName
+			}
+			err = c.createACLPolicyRoleAndBindingRule("connect-inject", injectRules, consulDC, primaryDC, globalToken, primary, componentAuthMethodName, serviceAccountName, consulClient)
```

Comment:
we could also just pass the `globalComponentAuthMethodName` here and avoid the `if` above:
```suggestion
			err = c.createACLPolicyRoleAndBindingRule("connect-inject", injectRules, consulDC, primaryDC, globalToken, primary, globalComponentAuthMethodName, serviceAccountName, consulClient)
```

Also do we still need to pass `globalToken` and `localToken`? If we do, maybe we should rename those since we're not creating tokens and it reads a bit confusing IMO

Reply:
renamed these to `globalPolicy` and `localPolicy`